program/src/api/rpc.rs

```diff
@@ -0,0 +1,126 @@
+//! gear api rpc methods
+#![allow(clippy::too_many_arguments)]
+use crate::{
+    api::{types::GasInfo, Api},
+    result::Result,
+};
+use gear_core::ids::{CodeId, MessageId, ProgramId};
+use std::sync::Arc;
+use subxt::{
+    rpc::{rpc_params, ClientT},
+    sp_core::H256,
+    RpcClient,
+};
+
+impl Api {
+    /// get rpc client
+    pub fn rpc(&self) -> Arc<RpcClient> {
+        self.client.rpc().client.clone()
+    }
+
+    /// gear_calculateInitCreateGas
+    pub async fn calculate_create_gas(
+        &self,
+        origin: H256,
+        code_id: CodeId,
+        payload: Vec<u8>,
+        value: u128,
+        allow_other_panics: bool,
+        at: Option<H256>,
+    ) -> Result<GasInfo> {
+        self.rpc()
+            .request(
+                "gear_calculateInitCreateGas",
+                rpc_params![
+                    origin,
+                    H256(code_id.into()),
+                    hex::encode(payload),
+                    u64::try_from(value).unwrap_or(u64::MAX),
+                    allow_other_panics,
+                    at
+                ],
+            )
+            .await
+            .map_err(Into::into)
+    }
+
+    /// gear_calculateInitUploadGas
+    pub async fn calculate_upload_gas(
+        &self,
+        origin: H256,
+        code: Vec<u8>,
+        payload: Vec<u8>,
+        value: u128,
+        allow_other_panics: bool,
+        at: Option<H256>,
+    ) -> Result<GasInfo> {
+        self.rpc()
+            .request(
+                "gear_calculateInitUploadGas",
+                rpc_params![
+                    origin,
+                    hex::encode(code),
+                    hex::encode(payload),
+                    u64::try_from(value).unwrap_or(u64::MAX),
+                    allow_other_panics,
+                    at
+                ],
+            )
+            .await
+            .map_err(Into::into)
+    }
+
+    /// gear_calculateHandleGas
+    pub async fn calculate_handle_gas(
+        &self,
+        origin: H256,
+        destination: ProgramId,
+        payload: Vec<u8>,
+        value: u128,
+        allow_other_panics: bool,
+        at: Option<H256>,
+    ) -> Result<GasInfo> {
+        self.rpc()
+            .request(
+                "gear_calculateHandleGas",
+                rpc_params![
+                    origin,
+                    H256(destination.into()),
+                    hex::encode(payload),
+                    u64::try_from(value).unwrap_or(u64::MAX),
```

Comment:
do we really can't throw here u128? use [u8; 16] instead for rpc calls also

Reply:
we never used u64 for value
If I'm not mistaken, rpc params macro wasn't able to use it for some reason